Assets/SEE/Controls/Actions/MarkAction.cs

```diff
@@ -0,0 +1,164 @@
+using SEE.Utils.History;
+using System.Collections.Generic;
+using UnityEngine;
+using SEE.GO;
+using SEE.Utils;
+using SEE.Game.Scenemanipulation;
+using SEE.Net.Actions;
+using SEE.Audio;
+using SEE.Game.SceneManipulation;
+
+namespace SEE.Controls.Actions {
+    public class MarkAction : AbstractPlayerAction
+    {
+        /// <summary>
+        /// Returns a new instance of <see cref="MarkAction"/>.
+        /// </summary>
+        /// <returns>new instance</returns>
+        public static IReversibleAction CreateReversibleAction()
+        {
+            return new MarkAction();
+        }
+
+        /// <summary>
+        /// Returns the <see cref="ActionStateType"/> of this action.
+        /// </summary>
+        /// <returns><see cref="ActionStateTypes.Mark"/></returns>
+        public override ActionStateType GetActionStateType()
+        {
+            return ActionStateTypes.Mark;
+        }
+
+        /// <summary>
+        /// If the user clicks with the mouse hitting a game object representing a graph node,
+        /// will be marked. A sphere will appear above the marked node.
+        /// <see cref="IReversibleAction.Update"/>.
+        /// </summary>
+        /// <returns>true if completed</returns>
+        public override bool Update()
+        {
+            bool result = false;
+
+            // FIXME: Needs adaptation for VR where no mouse is available.
+            if (Input.GetMouseButtonDown(0)
+                && Raycasting.RaycastGraphElement(out RaycastHit raycastHit, out GraphElementRef _) == HitGraphElement.Node)
+            {
+                // the hit object is the parent in which to create the sphere
+                GameObject parent = raycastHit.collider.gameObject;
+                Vector3 position = parent.transform.position;
+                Vector3 scale = parent.transform.lossyScale;
+                // create or delete the sphere
+                addedSphere = GameNodeMarker.CreateOrDeleteSphere(parent, scale);
```

Comment:
Why passing `scale` as a parameter? Position and scale can be retrieved by `GameNodeMarker.CreateOrDeleteSphere`.